src/shared/lib/markDownRules.tsx

```diff
@@ -418,12 +431,28 @@ const markDownRules: RenderRules = {
 
     const isCached = !!CacheManager.entries[node.attributes.src];
 
+    if (!imageSize.width || !imageSize.height) {
+      const imageDimensions = getImageSize(src);
+
+      if (!imageDimensions) {
+        throw Error(
+          `[markDownRules.image]: No image dimension found for url: ${src}`,
+        );
+      }
+
+      const { width, height } = imageDimensions;
+      imageSize = {
+        width: Number(width),
+        height: Number(height),
+      };
```

Comment:
`getImageSize` already returns a `number`, you could instead assign the return value directly